### PR TITLE
Improve manual test case docu

### LINKED ISSUE #640: Test Protocol
There have been some unclarities about where and how to write the testcases.
And a bullet point list is probably better than having the markdown table.

For comparison:

moving from this:
```
| Test Case | Description   | Preconditions   | Test Steps              | Expected Result | Actual Result | Test Result | Notes  |
| --------- | ------------- | --------------- | ----------------------- | --------------- | ------------- | ----------- | ------ |
| TC-001    | [Description] | [Preconditions] | 1. [Step 1 description] | [Expected]      | [Actual]      | ✔️          | [Note] |
|           |               |                 | 2. [Step 2 description] |                 |               |             |        |
|           |               |                 | ...                     |                 |               |             |        |
|           |               |                 |                         |                 |               |             |        |
| TC-002    | [Description] | [Preconditions] | 1. [Step 1 description] | [Expected]      | [Actual]      | ❌          | [Note] |
|           |               |                 | 2. [Step 2 description] |                 |               |             |        |
|           |               |                 | ...                     |                 |               |             |        |
|           |               |                 |                         |                 |               |             |        |
| TC-003    | [Description] | [Preconditions] | 1. [Step 1 description] | [Expected]      | [Actual]      | ⚠️          | [Note] |
|           |               |                 | 2. [Step 2 description] |                 |               |             |        |
|           |               |                 | ...                     |                 |               |             |        |
```

to something like this for example:
```
# TC-001 - Use Case: Login

- Test Steps:
  TODO: Fill out the test steps manually
- Expected Result:
  - [ ]
- Actual Result:
- Test Result:

# TC-002 - Login (Negative)

- Test Steps:
  There is an error in the app's authentication process.
  In this case, the app displays an error message to the user and tells them to try again.
- Expected Result: Error message is displayed
- Actual Result:
- Test Result:

# TC-003 - Use Case: Translation

- Test Steps:
  TODO: Fill out the test steps manually
- Expected Result:
  - [ ] The app's interface and text are displayed in the selected language.
- Actual Result:
- Test Result:
```

The general idea of a manual test is, to extend the "blueprint" of the tests(which is called the protocol) until shortly before a release, and then perform the manual test(the report).

So when writing a new test case you dont need to add the header (tester, testdate, commit hash, etc) and footer (error analysis) to your tests, since you already tested it multiple times during developing it and it should obviously work.

I will soon open a PR for these changes and make the process clearer.

### PR DESCRIPTION
**Pull request by 4ydan**
This should improve the readability of manual tests.
It should additionally make it clearer how they are written and performed.
I integrated plant_search test cases into the first protocol.
Closes #640 

## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: doc/tests/protocols/0.4_stable/protocol.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# Manuel Test Protocol
```
Comment (badnames) [inline doc/tests/protocols/0.4_stable/protocol.md:1]:
```suggestion
# Manual Test Protocol
```

File: doc/tests/protocols/plant_search.md
Comment (badnames) [inline doc/tests/protocols/plant_search.md:1]: Would it also be possible to translate this test protocol, so we don't have to write it all over again?

**Review by badnames — APPROVED** [review]
Looks good, just two minor nitpicks.

File: doc/tests/protocols/plant_search.md
Comment (4ydan) [reply doc/tests/protocols/plant_search.md:1]: Yes, I already did that. Its in [0.4_stable/protocol.md](https://github.com/ElektraInitiative/PermaplanT/pull/642/files#diff-2299e730f0ac9ff7bc13e0600852373af664a3f5dab1d4b9c99e98375b279aba)

File: doc/tests/protocols/0.4_stable/protocol.md (hunk 1 above)
Comment (4ydan) [reply doc/tests/protocols/0.4_stable/protocol.md]: oh lol, thx

File: doc/tests/protocols/plant_search.md
Comment (badnames) [reply doc/tests/protocols/plant_search.md:1]: Oh ok sorry I didn't make the connection.
I should probably go to sleep :sweat_smile:.

Comment (4ydan) [reply doc/tests/protocols/plant_search.md:1]: Hahhaa, dont worry.

File: doc/tests/protocols/0.4_stable/protocol.md
Comment (markus2330) [inline doc/tests/protocols/0.4_stable/protocol.md:1]: What means 0.4_stable? Do you want to add all manual tests in one file?

Comment (4ydan) [reply doc/tests/protocols/0.4_stable/protocol.md:1]: We can either have a protocol and report for each test or one protocol and multiple reports. We should probably start with only one protocol, I will correct it.

**Pull request merged**